Api/Services/VisitService.cs

```diff
@@ -139,4 +144,114 @@ public async Task<Result<VisitSummaryVM>> CreateVisitAsync(CreateVisitRequest re
             Deposit = visit.Deposit
         };
     }
+
+    /// <summary>
+    /// Reject a visit request as resturant owner or employee
+    /// </summary>
+    /// <param name="visitId">ID of the event</param>
+    /// <param name="currentUser">Current user for permission checking</param>
+    [ErrorCode(nameof(visitId), ErrorCodes.NotFound, "Event not found")]
+    [ErrorCode(nameof(visitId), ErrorCodes.UserAlreadyRejected, "User already considered")]
+    [ErrorCode(nameof(currentUser), ErrorCodes.UserAlreadyRejected, "User not qualified to reject")]
+    public async Task<Result> ApproveVisitRequestAsync(int visitId,User currentUser)
+    {
+        var visitFound = await context.Visits
+            .FirstOrDefaultAsync(e => e.VisitId == visitId);
+
+        if (visitFound == null)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = nameof(visitId),
+                ErrorMessage = "Event not found",
+                ErrorCode = ErrorCodes.NotFound
+            };
+        }
+
+        var result1 = await authorizationService.VerifyOwnerRole(visitFound.RestaurantId, currentUser);
+        var result2 = await authorizationService.VerifyRestaurantHallAccess(visitFound.RestaurantId, currentUser.Id);
+
+        if(result1.IsError && result2.IsError)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = nameof(currentUser.Id),
```

Comment:
```suggestion
                PropertyName = null,
```
`nameof(currentUser.Id)` to `"Id"`, to nic nie mówi klientowi, a `null` jest używane w wielu końcówkach dla zaznaczenia błędu generalnie związanego z requestem